### ISSUE TITLE
Tune function

### ISSUE THREAD

**Issue by MichielStock** [CenterForStatistics-UGent/xnet#33]
The `tune` function should ideally also directly be used to tune

- [x] add field with bestlambdas, containing the pair that minimizes the loss
- [x] ideally automatically retrain the model with this best regularization or return trained model.

**Comment by JoFAM** [comment]
Do I change this in the tune function, or do I add a function `retrain()` that gives a retrained model based on the tune function? I want to keep the possibility to access the matrix with the loss function results.

**Comment by MichielStock** [comment]
So, `tune()` best returns a tuned model:

- fitted with the combination of hyperparameters that minimize the loss
- this model has the following additional attributes:
   - hyperparameter grids for tuning
   - matrix with the results of the combinations
   - best lambdas
   - loss used
   - exclusion used

**Comment by JoFAM** [comment]
`tune` function is now rewritten to output a class `tskrrTune`. It's functional, but you have to extract all info from the object using the `@` operator for S4 objects for now.

Still to do:

 - [ ] write unit tests to check correct processing of input
 - [ ] write unit tests to check correct output
 - [ ] write getters for the different slots
    * [ ] lambda grid
    * [ ] loss
    * [ ] loss matrix
    * [ ] loss function
 - [ ] write a `loss()` method that can directly calculate the loss on a non-tuned `tskrr` object. For a `tskrrTune` class this would just return the loss with the best lambdas.
 - [ ] Make classes `tskrrTuneHeterogenous` and `tskrrTuneHomogenous` that inherit from both `tskrrTune` and the respective class `tskrrHeterogenous` and `tskrrHomogenous`. This way I won't have to provide methods for every single function to make the `tskrrTune` work as a model where needed. 
    * [ ] write class definitions (is easy, there's no difference in slots, just inheritance)
    * [ ] split up validity code for this
    * [ ] adjust the output of the `tune` function.
    * [ ] test !!!!!!

**Comment by JoFAM** [comment]
All finished in commit a138b51 

@MichielStock Currently loss_mse is used as the default. Maybe it makes more sense to use loss_auc? Depends on the values in Y probably (for continuous values MSE seems more appropriate, for 0/1 values I'd go for auc then). Options : 

 1. leave as is (loss_mse default)
 2. set loss_auc as the default
 3. remove the default so the user has to specify which loss function to use
 4. try to guess which one is most appropriate

Your choice? If this is OK, I can close the issue and pull everything into the master.

**Comment by JoFAM** [comment]
loss_mse stays default.
